Improved the aeson package to handle some border cases (#48)
ludat/conferer: docs/sources/json.md

Note that:

* Since conferer's values always are string, the `true` becomes `"true"`, conferer's
    FromConfig knows how to interpret this properly though (same as numbers and null).
* We traverse the tree until we find a primitive value.
* We don't define keys for nested values like `"some"` in this case.
* For list we define a key with its index and they can nest other values.
* The json file may only contain valid keys (lowercase ascii and numbers)
  otherwise an exception will be thrown.

### Special `_self` key

> Conferer allows setting some key and at the same time more specifici keys at the same

Reviewer: ```suggestion
Conferer allows setting some key and at the same time more specific keys at the same
```

* For list we define a key with its index and they can nest other values.
* All `Key`s are lowercase so this source ignores uppercase properties.

> ### Magic `_self` key

Reviewer: hahaha even tho I like the "Magic" I think it might be an offputing name for someone using the library 😅 . What about **Special** _self key?

> Conferer allows setting some key and at the same time more specifici keys at the same
> time.

Reviewer: ```suggestion

Conferer allows setting a value for a key and for more specific keys at the same time.
```

what about something like `subkeys of that key` instead of `more specific keys`?

> For example setting:

Reviewer: ```suggestion
For example, the following setting is valid:
```

`somefile`=myfile.png
`somefile.extension`=jpg

> is possible and when parsing `File` this will result in `myfile.jpg` (more specific wins)

Reviewer: ```suggestion
and if parsed as a `File` will result in `myfile.jpg` (more specific wins)
```
